src/taipy/core/_orchestrator/_orchestrator_factory.py

```diff
@@ -49,48 +49,50 @@ def _build_dispatcher(cls, force_restart=False) -> Optional[_JobDispatcher]:
         elif Config.job_config.is_development:
-            return cls.__build_development_job_dispatcher()
+            cls.__build_development_job_dispatcher()
+        elif util.find_spec(cls._TAIPY_ENTERPRISE_MODULE) is not None:
+            cls.__build_enterprise_job_dispatcher(force_restart)
         else:
             raise ModeNotAvailable(f"Job mode {Config.job_config.mode} is not available.")
+        return cls._dispatcher
 
     @classmethod
     def _remove_dispatcher(cls) -> Optional[_JobDispatcher]:
-        if isinstance(cls._dispatcher, _StandaloneJobDispatcher) and not isinstance(
-            cls._dispatcher, _DevelopmentJobDispatcher
-        ):
+        if cls._dispatcher is not None and not isinstance(cls._dispatcher, _DevelopmentJobDispatcher):
             cls._dispatcher.stop()
-
         cls._dispatcher = None
-
         return cls._dispatcher
 
     @classmethod
-    def __build_standalone_job_dispatcher(cls, force_restart=False) -> Optional[_JobDispatcher]:
-        if isinstance(cls._dispatcher, _StandaloneJobDispatcher) and not isinstance(
-            cls._dispatcher, _DevelopmentJobDispatcher
-        ):
+    def __build_standalone_job_dispatcher(cls, force_restart=False):
+        if isinstance(cls._dispatcher, _StandaloneJobDispatcher):
             if force_restart:
                 cls._dispatcher.stop()
             else:
-                return None
+                return
 
         if util.find_spec(cls._TAIPY_ENTERPRISE_MODULE) is not None:
             cls._dispatcher = _load_fct(
-                cls._TAIPY_ENTERPRISE_CORE_DISPATCHER_MODULE, cls.__STANDADLONE_JOB_DISPATCHER_TYPE
+                cls._TAIPY_ENTERPRISE_CORE_DISPATCHER_MODULE, cls.__TAIPY_ENTERPRISE_BUILD_DISPATCHER_METHOD
             )(cls._orchestrator)
         else:
             cls._dispatcher = _StandaloneJobDispatcher(cls._orchestrator)  # type: ignore
-
         cls._dispatcher.start()  # type: ignore
-        return cls._dispatcher
 
     @classmethod
-    def __build_development_job_dispatcher(cls) -> _JobDispatcher:
-        if isinstance(cls._dispatcher, _StandaloneJobDispatcher) and not isinstance(
-            cls._dispatcher, _DevelopmentJobDispatcher
-        ):
+    def __build_development_job_dispatcher(cls):
+        if isinstance(cls._dispatcher, _StandaloneJobDispatcher):
             cls._dispatcher.stop()
         cls._dispatcher = _DevelopmentJobDispatcher(cls._orchestrator)  # type: ignore
-        return cls._dispatcher
+
+    @classmethod
+    def __build_enterprise_job_dispatcher(cls, force_restart=False):
+        cls._dispatcher = _load_fct(
+            cls._TAIPY_ENTERPRISE_CORE_DISPATCHER_MODULE, cls.__TAIPY_ENTERPRISE_BUILD_DISPATCHER_METHOD
+        )(cls._orchestrator, force_restart)
+        if cls._dispatcher:
+            cls._dispatcher.start()
+        else:
+            raise ModeNotAvailable(f"Job mode {Config.job_config.mode} is not available.")
```

Comment:
Should this be moved to enterprise side?

Because I'm not sure if all dispatcher type returned by enterprise can be `.start()`.

Reply:
Yes, the enterprise dispatcher can be started. I would keep it here.